tardis/energy_input/transport/gamma_ray_grid.py

```diff
@@ -34,18 +34,29 @@ def calculate_distance_radial(photon, r_inner, r_outer):
         photon.location, photon.direction, r_outer
     )
 
-    final_position_inner_1 = photon.location + photon.direction * inner_1
-    final_position_inner_2 = photon.location + photon.direction * inner_2
-    final_position_outer_1 = photon.location + photon.direction * outer_1
-    final_position_outer_2 = photon.location + photon.direction * outer_2
+    final_position_inner_1 = np.ascontiguousarray(
+        photon.location + photon.direction * inner_1
+    )
+    final_position_inner_2 = np.ascontiguousarray(
+        photon.location + photon.direction * inner_2
+    )
+    final_position_outer_1 = np.ascontiguousarray(
+        photon.location + photon.direction * outer_1
+    )
+    final_position_outer_2 = np.ascontiguousarray(
+        photon.location + photon.direction * outer_2
+    )
+
+    # Ensure photon.direction is contiguous for dot product operations
+    direction_contiguous = np.ascontiguousarray(photon.direction)
```

Comment:
Not sure why these arrays need to be forced contiguous, because I am not sure how they would end up in the Fortran style